src/main/java/org/jenkinsci/plugins/docker/workflow/FromFingerprintStep.java

```diff
@@ -108,8 +115,12 @@ public String getToolName() {
                             continue;
                         }
                         if (line.startsWith("FROM ")) {
-                            fromImage = line.substring(5);
-                            continue;
+                            String[] parts = line.split("\\s+");
+                            // if target is set, we want to find the line which matches "FROM x AS target"
```

Comment:
You should probably verify that parts[2] == "AS"

Reply:
Maybe... but https://docs.docker.com/engine/reference/builder/#from says there is no other valid format than `FROM x` or `FROM x AS y` so if `parts[2]` is anything other than `AS` I think Docker will be throwing a fit before Jenkins can parse the file. This is all a dirty hack regardless so I think it's OK to assume this part.